### DISCUSSION TITLE
ONNX export using ONNX 1.5 throws up problem in BatchNorm operator

### DISCUSSION THREAD

**Discussion by Mut1nyJD** [apache/incubator-mxnet#15536] [Q&A]
## Description
I installed the latest ONNX version 1.5 and now trying to export a standard model (resnet18v2) from the gluoncv modelzoo is causing a validation error in onnx,onnx_cpp2py.export.checker

It is basically complaining about Unrecognized attribute: spatial for operator BatchNormalization

## Environment info (Required)
Windows10, CUDA9.2, Python 3.5
mxnetcu92-mkl 1.4.0.post0
gluoncv0.5.0b20190326
onnx1.5.0

[a comment by an automated account is not shown]

**Comment by braindotai** [comment]
This [page](https://mxnet.incubator.apache.org/api/python/contrib/onnx.html) tells that only onnx version 1.2.1 is supported. After downgrading onnx from 1.5.0 to 1.2.1, everything should work fine.

**Comment by Mut1nyJD** [comment]
@braindotai Yes it resolves the export problem however some ONNX inference engines are very picky and do not accept this spatial notation in batchnorm. So it does not really solve the problem. 

Are there any plans to move to ONNX1.5 (opspec 12) as it finally opens up the ONNX to more complex models like e.g. object classification

**Comment by braindotai** [comment]
I'm quite note sure about that, members of MXNet might help you out.

**Comment by frankfliu** [comment]
@mxnet-label-bot add [ONNX, question]

**Comment by vandanavk** [comment]
MXNet currently supports ONNX v1.3. The documentation is updated in PR https://github.com/apache/incubator-mxnet/pull/15914

**Comment by ChaiBapchya** [comment]
It is fixed with https://github.com/apache/incubator-mxnet/pull/17711
So exporting model using mxnet build from source or nightly binaries of 2.0 should resolve this issue. However, the issue will persist on mx1.4/1.5.1/1.6.0 pypi wheels
@szha

**Comment by szha** [comment]
@ChaiBapchya thanks Chai. shall we backport the patch to 1.x?

**Comment by ChaiBapchya** [comment]
Yes. Here you go: https://github.com/apache/incubator-mxnet/pull/18846
However, I'm worried folks consuming MXNet out of PyPi binaries release versions [v1.5.1/v1.6] still won't get this fix. They'd have to either build from source / use nightly binary / manually remove the line from installed python package.
